Aplib.Core/Desire/Goals/Goal.cs

```diff
@@ -51,70 +56,143 @@ public class Goal<TBeliefSet> : IGoal<TBeliefSet>, IDocumented
         /// </summary>
         public ITactic<TBeliefSet> Tactic { get; }
 
-        /// <inheritdoc />
+        /// <summary>
+        /// Gets the completion status of the goal.
+        /// This value may need to be updated first using the <see cref="UpdateStatus"/> method.
+        /// </summary>
+        /// <seealso cref="UpdateStatus"/>
         public CompletionStatus Status { get; protected set; }
```

Comment:
Maybe it's an idea to call UpdateStatus in the getter here before returning it?

Reply:
Yes that's a good idea actually.

Reply:
I think actually we might remove the `UpdateStatus` method completely and just put it in the getter.

Reply:
And also from goal structures, for consistency.